### ISSUE TITLE
Robustly parse strings that represent NaN and Infinity

### ISSUE THREAD

**Issue by cwardgar** [Unidata/thredds#569]
From the [mailing list](http://www.unidata.ucar.edu/mailing_lists/archives/thredds/2016/msg00145.html):

In NetCDF-Java and THREDDS, we currently use [Double.valueOf](https://docs.oracle.com/javase/8/docs/api/java/lang/Double.html#valueOf-java.lang.String-) (or similar) to parse strings as floats. As the Javadoc states, only the literal string `"NaN"` is interpreted as Not-a-Number and only the literal string `"Infinity"` is interpreted as infinity. We ought to be more lenient. At the least, do case-insensitive matching and accept the `"inf"` string.

**Comment by DennisHeimbigner** [comment]
Agreed, although finding all the relevant places may  be a bit of a chore.
